lib/angular/template/user-template.tpl.html

```diff
@@ -0,0 +1,35 @@
+<div ng-if="loading">
+  <h1>Loading User List...</h1>
+</div>
+
+<div class="user_list" ng-if="!loading">
+  <h1>List Of Users</h1>
+
+  <ul>
+    <li ng-repeat="user in users">
+      {{ user.name }}
+    </li>
+  </ul>
+</div>
+
+<div class="user_form">
+  <h1>Add A User</h1>
+
+  <label for="inputName" class="">Name</label>
+  <input type="text"
+         name="inputName"
+         ng-model="newUser.name"
+         ng-required="true"
+  ></input>
+
+  <label for="inputAddress" class="">Address</label>
+  <input type="text"
+         name="inputAddress"
+         ng-model="newUser.address"
+         ng-required="true"
+  ></input>
+  <button ng-click="addUser()">Add New User</button>
```

Comment:
Could you use a [`<form>`](https://docs.angularjs.org/api/ng/directive/form) instead of a  `<div>` and [`ngSubmit`](https://docs.angularjs.org/api/ng/directive/form) instead of `ngClick`?